client/src/services/bms/api.ts

```diff
@@ -51,16 +61,25 @@ function parseBmsErrors(data: object): Array<BmsError> {
   return [];
 }
 
-async function login(query: LoginQueryData): Promise<BmsResponse> {
+async function wrapQuery(queryFunc: () => Promise<BmsResponse>): Promise<BmsResponse> {
```

Comment:
A lot better than before, but would be maybe even more better with decorators, not necessary now, but to keep in mind in the future.